repository/kafka/operator/templates/statefulset.yaml

```diff
@@ -67,6 +67,8 @@ spec:
               value: "{{ .Params.CLIENT_PORT_ENABLED }}"
             - name: KAFKA_BROKER_PORT
               value: "{{ .Params.BROKER_PORT }}"
+            - name: KAFKA_BROKER_PORT_TLS
+              value: "{{ .Params.BROKER_PORT_TLS }}"
```

Comment:
we should also use it in `containerPort: {{ .Params. BROKER_PORT_TLS }}`  when `{{ if eq .Params.TRANSPORT_ENCRYPTION_ENABLED "true" }}` is true

and also change it in `service.yaml` and use it instead of `port: {{ .Params.BROKER_PORT }}` when TLS encryption is enabled

Reply:
As discussed in the last team sync, the `BROKER_PORT` is only used along with `BROKER_PORT_TLS` **only when** `TRANSPORT_ENCRYPTION_ALLOW_PLAINTEXT` is enabled. It is disabled by default. In future release, if users want they can use  `CLIENT_PORT` as a fixed `PLAINTEXT` listener to allow unrestricted access to the cluster.